android/app/src/main/java/hr/foi/air/wattsup/screens/CardScreen.kt

```diff
@@ -0,0 +1,366 @@
+import android.content.Context
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.gestures.snapping.rememberSnapFlingBehavior
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ElevatedButton
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import hr.foi.air.wattsup.R
+import hr.foi.air.wattsup.network.models.Card
+import hr.foi.air.wattsup.network.models.TokenManager
+import hr.foi.air.wattsup.ui.component.CircleButton
+import hr.foi.air.wattsup.ui.component.TopAppBar
+import hr.foi.air.wattsup.viewmodels.CardViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
```

Comment:
No need to supress warning for unused scaffold padding, since it is better to use it anyway to ensure correct padding for all device screens. `@SuppressLint` can be removed and instead of calling `CardView(viewModel, onAddCard)`, change signature to:

```
fun CardView(
    viewModel: CardViewModel,
    onAddCard: () -> Unit,
    modifier: Modifier = Modifier,
    context: Context = LocalContext.current,
)
```

 and call `CardView(viewModel, onAddCard, Modifier.padding(it))`. In CardView's Column, instead of

```
Column(
        modifier = Modifier
            .padding(top = 50.dp)
            .fillMaxSize(),
    )
```

call (small `m` in `modifier` and remove top padding)

```
Column(
        modifier = modifier
            .fillMaxSize(),
    ) 
```

That way, scaffold's padding will be applied to this column without having to guesstimate padding.

Reply:
Cool, done